raiden/channel/balance_proof.py

```diff
@@ -89,6 +89,29 @@ def locked(self):
             for lock in alllocks
         )
 
+    def register_balanceproof_without_lock(self, balance_proof, lock):
+        lockhashed = sha3(lock.as_bytes)
+
+        if not isinstance(balance_proof, BalanceProofState):
+            raise ValueError('balance_proof must be a BalanceProof instance')
+
+        if not self.is_known(lock.hashlock):
+            raise ValueError('hashlock is not registered')
+
+        leafs = self.unclaimed_merkletree()
+        leafs.remove(lockhashed)
+        new_locksroot = Merkletree(leafs).merkleroot
+
+        if balance_proof.locksroot != new_locksroot:
+            raise InvalidLocksRoot(new_locksroot, balance_proof.locksroot)
+
+        if lock.hashlock in self.hashlocks_to_pendinglocks:
```

Comment:
Can any of these `del` fail because of concurrency? If so, I would suggest a `try-finally` scheme, to ensure we set the balance_proof.

Reply:
No, we are using cooperative tasks, this code does no IO so a context-switch cannot happen after the `is_know` check and before the delete.